src/components/form/HdInputPhone.vue

```diff
@@ -0,0 +1,368 @@
+            <p class="option-country__name-and-code">
+              <span class="option-country__name">{{t.COUNTRIES[country.code]}}</span>
+              <span class="option-country__dial-code">{{country.dial_code}}</span>
+            </p>
+          </button>
+          <hr v-if="index === preferredCountries.length - 1" class="phone-input__dropdown__divider" />
+        </li>
+      </template>
+    </ul>
+    <HdInputFormatter
+      ref="input"
+      type="tel"
+      name="phoneNumber"
+      label="Telefonnumer"
+      v-model="phoneNumber"
+      class="phone-input__field"
+      placeholder="+55 555 555 5555"
+      :custom-rules="[rule]"
+      :formatter="phoneFormatter"
+      @input="handleInputEvent"
+    />
+  </div>
+</template>
+
+<script>
+/* eslint-disable import/no-extraneous-dependencies */
+import { action } from '@storybook/addon-actions';
+import HdInputFormatter from 'homeday-blocks/src/components/form/HdInputFormatter.vue';
+import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
+import { getMessages } from 'homeday-blocks/src/lang';
+import { formatPhoneNumber } from 'homeday-blocks/src/services/utils';
+import { smallArrow as smallArrowIcon } from 'homeday-assets';
+import countryCodes from 'country-codes-list';
+import formField from './formFieldMixin';
+
+const COUNTRY_PHONE_CODES = countryCodes
+  .customArray({ code: '{countryCode}', dial_code: '+{countryCallingCode}' });
+
+const MAX_DIGITS_DIAL_CODE = 5;
+const MIN_DIGITS_DIAL_CODE = 3;
+const ARROW_UP_KEY_CODE = 38;
+const ARROW_DOWN_KEY_CODE = 40;
+const ESC_KEY_CODE = 27;
+
+const ALPHABET_UPPERCASE_CODES = [65, 90];
+const ALPHABET_LOWERCASE_CODES = [97, 122];
+
+const isAlphabetKey = (key) => {
+  const [A, Z] = ALPHABET_UPPERCASE_CODES;
+  const [a, z] = ALPHABET_LOWERCASE_CODES;
+  return (key >= A && key <= Z) || (key >= a && key <= z);
+};
+
+export default {
+  name: 'HdInputPhone',
+  mixins: [
+    formField,
+  ],
+  inheritAttrs: false,
+  components: {
+    HdInputFormatter,
+    HdIcon,
+  },
+  props: {
+    defaultCountry: {
+      type: String,
+      default: 'DE',
+    },
+    preferredCountries: {
+      type: Array,
+      default: () => ([]),
+    },
+    value: {
+      type: String,
+      default: '',
+    },
+    name: {
+      type: String,
+      required: true,
+    },
+    lang: {
+      type: String,
+      default: 'de',
+    },
+  },
+  data() {
+    return {
+      showDropdown: false,
+      phoneNumber: COUNTRY_PHONE_CODES.find((country) => country.code === this.defaultCountry)?.dial_code,
+      selectedCountry: COUNTRY_PHONE_CODES.find((country) => country.code === this.defaultCountry),
+      focusedCountry: null,
+      rule: {
+        validate: (value) => {
+          const hasMoreThan8Digits = value.length >= 8;
+          const hasPlusDigit = value.includes('+');
+          const isValid = hasPlusDigit && hasMoreThan8Digits;
+          return isValid;
```

Comment:
Wow 🤓  I got inspired by your approach to managing the if condition here. Make total sense!

Reply:
😅 thanks!